federation-js/src/service/printSupergraphSdl.ts

```diff
@@ -321,54 +360,81 @@ export function printWithReducedWhitespace(ast: ASTNode): string {
 }
```

Comment:
up a few lines^^

A couple years ago [at our request](https://github.com/graphql/graphql-js/pull/1802) GraphQL gained a `stripIgnoredCharacters` function that maps a string to a string removing unnecessary characters (including comments) in a more rigorous way. It made it into 14.3 so it's in all versions our peer deps ask for. We should use that instead of the `.replace`!

It looks like the repo passes tests modulo a couple trivially-updated snapshot tests if you just replace this function with `return stripIgnoredCharacters(print(ast))` today. Perhaps this is too stressful to do as part of this big PR (eg, it may slightly affect query plans in trivial ways that might confuse our validation maybe?) but can you at least add a comment suggesting we try the change at some point?

Reply:
(specifically this matters because the current code mangles whitespace in string literals in arguments, and the upstream function doesn't)